### PR TITLE
[release-0.53]: Fix: Align Reenlightenment flows between converter.go and template.go

### PR DESCRIPTION
**Pull request by iholder101**
**What this PR does / why we need it**:
Manual backport of https://github.com/kubevirt/kubevirt/pull/8798. As a dependency, I've also backported https://github.com/kubevirt/kubevirt/pull/8793.

The reasons for the manual backport are mostly minor issues.

**Release note**:

```release-note
Fix: Align Reenlightenment flows between converter.go and template.go
```

### CODE REVIEW

**Comment by iholder101** [comment]
/cc @enp0s3
/cc @Barakmor1

can you please take a look?

**Comment by Barakmor1** [comment]
Please divide the first commit to the original commit and another commit with your changes so it will be easier to review.

**Comment by iholder101** [comment]
> Please divide the first commit to the original commit and another commit with your changes so it will be easier to review.

done

File: pkg/virt-controller/services/template.go [hunk 1]
```diff
@@ -1877,10 +1874,11 @@ func appendUniqueImagePullSecret(secrets []k8sv1.LocalObjectReference, newsecret
 // This includes the memory needed for the guest and memory
 // for Qemu and OS overhead.
 //
-// The return value is overhead memory quantity
+// # The return value is overhead memory quantity
 //
 // Note: This is the best estimation we were able to come up with
-//       and is still not 100% accurate
+//
+//	and is still not 100% accurate
```
Comment (Barakmor1) [inline pkg/virt-controller/services/template.go:1881]: nit: i think that these are leftovers

**Comment by Barakmor1** [comment]
/lgtm 
Thank you!

**Pull request merged**